File: guides/micronaut-metrics/kotlin/src/main/kotlin/example/micronaut/crypto/BitcoinPrice.kt
package example.micronaut.crypto

import io.micronaut.serde.annotation.Serdeable

@Serdeable // <1>
class BitcoinPrice(private val data: Data) {

    val price: Float
        get() = data.price

    @Serdeable // <1>
    class Data(val price: Float)
}

File: guides/micronaut-metrics/python/src/example/micronaut/crypto/bitcoin_price.py
from dataclasses import dataclass

from micronaut.serde.annotation import Serdeable


@dataclass
@Serdeable  # <1>
class Data:
    price: float


@dataclass
@Serdeable  # <1>
class BitcoinPrice:
    data: Data

    @property
    def price(self) -> float:
        return self.data.price
